src/windows/osd.c

```diff
@@ -402,65 +402,104 @@ ssize_t sendmsg(SOCKET sd, struct msghdr *msg, int flags)
 	goto fn_complete;
 }
 
+void sort_list_of_adapters(PIP_ADAPTER_ADDRESSES head)
+{
+	PIP_ADAPTER_ADDRESSES ptr = head, tmp = NULL, prev = NULL;
+	bool sorted;
+
+	if (!head)
+		return;
+
+	do {
+		sorted = false;
+		ptr = head;
+		for (ptr = head; ptr->Next; ptr = ptr->Next) {
+			if (ptr->Ipv4Metric < ptr->Next->Ipv4Metric) {
+				tmp = ptr;
+				ptr = tmp->Next;
+				tmp->Next = ptr->Next;
+				ptr->Next = tmp;
+				if (tmp == head)
+					head = ptr;
+				else
+					prev->Next = ptr;
+				sorted = true;
+			}
+			ptr = ptr->Next;
+		}
+	} while (sorted);	
```

Comment:
It wasn't obvious reading this the first time, but this is attempting to perform a bubble sort on the singly linked list.  It is incorrect.  The looping isn't quite right, as the for loop should iterate over a smaller and smaller portion of the list.  The swapping function (portion inside the if statement) isn't correct either.  It doesn't update the Next pointer for the item that comes before the two being swapped, which results in a corrupted list.